### DISCUSSION TITLE
FabricJS generating PNG in nodeJS, but sometimes images show as black rectangles. Cause and options to fix/handle?

### DISCUSSION THREAD

**Discussion by chau-a-nguyen** [fabricjs/fabric.js#8320] [Q&A]
We generate PNGs from the canvas in nodeJS, but usually when there are many images, sometimes some of the images just display as black rectangles. We assume those images timeout and so the black rectangles are there.

What is the cause? What are the timeouts (nodejs, express, fabricjs,... ) and values and configurations? What are the best options and workarounds? Thanks!

**Comment by ShaMan123** [comment]
You should open an issue and follow all the steps
For now add a reproduction.
Without it no one can guess

**Comment by ShaMan123** [comment]
But since you wrote images the first thing I would guess is that you didn't wait for them to finish loading
